draco/asp/hard.lp

```diff
@@ -141,3 +141,50 @@ violation(view_scale_conflict) :-
     entity(scale,V,S2),
     attribute((scale,channel),S1,C),
     attribute((scale,channel),S2,C).
+
+% @hard(shape_without_point)  Shape channel requires point mark.
+violation(shape_without_point) :-
+    attribute((mark,channel),M,shape),
+    not attribute((mark,type),M,point).
+
+% @hard(size_without_point_text) Size only works with some marks.
+violation(size_without_point_text) :-
+    attribute((mark,channel),M,size),
+    not attribute((mark,type),M,text),
+    not attribute((mark,type),M,point).
+
+% @hard(detail_without_agg)  Detail requires aggregation. Detail adds a field to the group by. Detail could also be used to add information to tooltips. We may remove this later.
+violation(detail_without_agg) :-
+    attribute((encoding,channel),E,detail),
+    not attribute((encoding,aggregate),E,_).
```

Comment:
Is this right? I thought detail needs aggregation on another channel in the same mark.